### ISSUE TITLE
Request - typed style<T> { } DSL

### ISSUE THREAD

**Issue by reubenfirmin** [rjaros/kvision#252]
Example:

```
style<Button> {
   ...
}
```

would be the equivalent of: 

```
addCssClass("my_widget")

//... and in the css

.my_widget button {
   ...
} 
```

or: 

```
val buttonStyle = Style() {
   // ...
} 

button {
    // ...
    addCssStyle(buttonStyle)
}

button {
   // ...
   addCssStyle(buttonStyle)
}
...
```

**Comment by rjaros** [comment]
The very similar DSL is already there:
```
            button("Test") {
                style {
                    width = 100.perc
                }
            }
```
isn't this the same?

**Comment by reubenfirmin** [comment]
I added one more example to the original. Suppose you have 5 or 10 buttons. You can declare a specific style for buttons and then add it to each, but that's a bit more verbose than the declarative css alternative.

**Comment by rjaros** [comment]
It's probably doable, but will require to map somehow every widget type to a html element name. It would probably add a few bytes (or kbytes?) to the bundle size. And would be problematic to keep all in sync with widgets in different, optional modules. I'll make some tests, but I'm not optimistic about the result :-(

**Comment by rjaros** [comment]
Unfortunately I haven't managed to create any useful solution for this typed style request. Performance and size impact is too big. But with some simple changes for the next release, you will be able to use the following code:
```
val mystyle = style {
    style("button") {
        colorName = Col.RED
        width = 300.px
    }
}

simplePanel(mystyle) {

    button("Button 1") {
    }

    button("Button 2") {
    }

}
```
Not exactly the same, but IMHO similar enough ;-)

**Comment by rjaros** [comment]
Fixed in 4.5.0
